users/views.py

```diff
@@ -1,3 +1,68 @@
-from django.shortcuts import render
+import json, bcrypt, jwt
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+from django.core.exceptions import ValidationError
+
+from users.models     import User
+from users.validators import validate_email, validate_password
+from my_settings      import SECRET_KEY, ALGORITHM
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            username        = data.get('username')
+            email           = data.get('email')
+            password        = data.get('password')
+            phone_number    = data.get('phone_number')
+            address         = data.get('address')
+            hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+
+            if validate_email(email)==None:
+                return JsonResponse({'messasge':'INVALID EMAIL'}, status=400) 
+            if validate_password(password)==None:
+                return JsonResponse({'messasge':'INVALID PASSWORD'}, status=400) 
+            if User.objects.filter(email = email).exists():
+                return JsonResponse({'messasge':'E-MAIL ALREADY EXISTED'}, status=400)
+
+            User.objects.create(
+                username     = username,
+                email        = email,
+                password     = hashed_password,
+                phone_number = phone_number,
+                address      = address,
+                point        = 100000,
+            )
+            return JsonResponse({'message':'SUCCESS'}, status=201) 
+        except KeyError:
+            return JsonResponse({'message':'KEY_ERROR'},status=400) 
+        except ValidationError as e:
+            return JsonResponse({'message' : e.message}, status = 400)
+
+class LogInView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            email           = data['email']
+            password        = data['password']
+            user            = User.objects.get(email = email)
+            hashed_password = user.password.encode('utf-8')
+            access_token    = jwt.encode({'user.id':user.id}, SECRET_KEY, ALGORITHM)
+
+            if not bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
+                return JsonResponse({'message' : 'INVALID_USER'}, status = 401)
+
+            if bcrypt.checkpw(password.encode('utf-8'), hashed_password):
+                
+                success_message = {
+                    'SUCCESS' : {
+                        'E-MAIL':email,
+                        'ACCESS_TOKEN':access_token,
+                    }
+                }
+            return JsonResponse(success_message, status=200)
```

Comment:
```suggestion                
            result = {
                'data' : {
                    'email' : email,
                    'access_token' : access_token,
                }
            }
        return JsonResponse(result, status=200)
```
- 첫번째 if에서 비밀번호 검증을 했기 때문에 또 다시 if로 검증하지 않아도 됩니다.
- json의 키는 보통 소문자로 사용합니다.
- 응답 데이터 구조는 제가 리뷰 남긴 구조로 하면 좋을 것 같습니다.

Reply:
바로 수정하겠습니다!